.github/workflows/release.yml

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+# See the file LICENSE for licensing terms.
+
+name: Release Teleporter
+
+on:
+  push:
+    tags:
+      - "v*.*.0"
+
+jobs:
+  build_and_upload_artifacts:
+    name: Build and Upload Teleporter Artifacts
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      
+      - name: Set Go version
+        run: |
+          source ./scripts/versions.sh
+          echo GO_VERSION=$GO_VERSION >> $GITHUB_ENV
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+            go-version: ${{ env.GO_VERSION }}
+
+      - name: Install Foundry
+        run: ./scripts/install_foundry.sh
+    
+      - name: Build Contracts
+        run: |
+          export PATH=$PATH:$HOME/.foundry/bin
+          cd contracts/
+          forge build
+
+      - name: Build changelog
+        id: build_changelog
+        uses: mikepenz/release-changelog-builder-action@v4
+        with:
+          failOnError: true
+          configuration: "./.github/changelog.json"
+          toTag: ${{ github.ref_name }}
+
+      - name: Create Artifacts
+        id: artifacts
+        run: |
+          go run utils/contract-deployment/contractDeploymentTools.go constructKeylessTx contracts/out/TeleporterMessenger.sol/TeleporterMessenger.json
+          
+          mv UniversalTeleporterDeployerTransaction.txt TeleporterMessenger_DeployerTransaction_${{ github.ref_name }}.txt
+          echo "tx=TeleporterMessenger_DeployerTransaction_${{ github.ref_name }}.txt" >> $GITHUB_OUTPUT
+          
+          mv UniversalTeleporterDeployerAddress.txt TeleporterMessenger_DeployerAddress_${{ github.ref_name }}.txt
+          echo "deployer=TeleporterMessenger_DeployerAddress_${{ github.ref_name }}.txt" >> $GITHUB_OUTPUT
+          
+          mv UniversalTeleporterMessengerContractAddress.txt TeleporterMessenger_ContractAddress_${{ github.ref_name }}.txt
+          echo "address=TeleporterMessenger_ContractAddress_${{ github.ref_name }}.txt" >> $GITHUB_OUTPUT
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        with:
+          name: ${{ github.ref_name }}
+          tag_name: ${{ github.ref_name }}
+          prerelease: false
+          body: ${{ steps.build_changelog.outputs.changelog }}
+          files: |
+                ${{ steps.artifacts.outputs.tx }}
+                ${{ steps.artifacts.outputs.deployer }}
+                ${{ steps.artifacts.outputs.address }}
```

Comment:
My understanding here is that there are 3 files that will be uploaded, named:
- `TeleporterMessenger_DeployerTransaction_${{ github.ref_name }}.txt`
- `TeleporterMessenger_DeployerAddress_${{ github.ref_name }}.txt`
- `TeleporterMessenger_ContractAddress_${{ github.ref_name }}.txt`

Could we simplify the workflow by setting those variables up top somehow, rather than renaming the files to them above, outputing the file names as other variable names, and then referencing that output here?

Reply:
I could get rid of one step here, by replacing lines 70-72 here with just the expected filenames, and then we could get rid of the 3 `echo` lines above. 

We could also modify the `ConstructKeylessTransaction` function to take an optional `version` parameter, and just make it write out the filenames we want in the first place, eliminating the 3 `mv` lines.

Reply:
> I could get rid of one step here, by replacing lines 70-72 here with just the expected filenames, and then we could get rid of the 3 echo lines above.

Yeah I think that'd be great 🙏 Bonus points if we can set the 3 file names as variables at the workflow level rather than having them in 2 place also, but I'm not sure how that can be configured myself